src/features/userProfile/state/userProfileState.ts

```diff
@@ -62,20 +65,25 @@ export const userProfileReducer = createReducer(initialState, {
     state.error = NewApiErr()
   },
   'API_USERPROFILE_SUCCESS': (state: UserProfileState, action) => {
-    state.profile = action.payload.data
+    state.profile = NewUserProfile(action.payload.data)
     state.loading = false
   },
   'API_USERPROFILE_FAILURE': (state: UserProfileState, action) => {
     state.loading = false
     state.error = action.payload.err
   },
 
+  USERPROFILE_SET: (state: UserProfileState, action: UserProfileAction) => {
+    state.profile = action.user
+  },
+
   'API_USERPROFILEDATASETS_REQUEST': (state: UserProfileState, action) => {
     state.datasets.loading = true
   },
   'API_USERPROFILEDATASETS_SUCCESS': (state: UserProfileState, action) => {
     state.datasets.results = action.payload.data
-    state.datasets.pageInfo = action.payload.pagination
+    state.datasets.pageInfo = action.payload.request.pageInfo
+    state.datasets.pageInfo.resultCount = state.datasets.results.length
```

Comment:
We are now using the `list` endpoint, that does not respond with the same kind of pageInfo. We are instead creating it here to match the expectations of the `UserProfile` page. Also, working on unifying our pagination, so this will probably change soon.